torchtext/experimental/datasets/raw/text_classification.py

```diff
@@ -0,0 +1,250 @@
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.DBpedia()
+    """
+
+    return _setup_datasets(*(("DBpedia",) + args), **kwargs)
+
+
+def YelpReviewPolarity(*args, **kwargs):
+    """ Defines YelpReviewPolarity datasets.
+
+    Create supervised learning dataset: YelpReviewPolarity
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.YelpReviewPolarity()
+    """
+
+    return _setup_datasets(*(("YelpReviewPolarity",) + args), **kwargs)
+
+
+def YelpReviewFull(*args, **kwargs):
+    """ Defines YelpReviewFull datasets.
+
+    Create supervised learning dataset: YelpReviewFull
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.YelpReviewFull()
+    """
+
+    return _setup_datasets(*(("YelpReviewFull",) + args), **kwargs)
+
+
+def YahooAnswers(*args, **kwargs):
+    """ Defines YahooAnswers datasets.
+
+    Create supervised learning dataset: YahooAnswers
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.YahooAnswers()
+    """
+
+    return _setup_datasets(*(("YahooAnswers",) + args), **kwargs)
+
+
+def AmazonReviewPolarity(*args, **kwargs):
+    """ Defines AmazonReviewPolarity datasets.
+
+    Create supervised learning dataset: AmazonReviewPolarity
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.AmazonReviewPolarity()
+    """
+
+    return _setup_datasets(*(("AmazonReviewPolarity",) + args), **kwargs)
+
+
+def AmazonReviewFull(*args, **kwargs):
+    """ Defines AmazonReviewFull datasets.
+
+    Create supervised learning dataset: AmazonReviewFull
+
+    Separately returns the training and test dataset
+
+    Arguments:
+        root: Directory where the datasets are saved. Default: ".data"
+
+    Examples:
+        >>> train, test = torchtext.experimental.raw_datasets.AmazonReviewFull()
+    """
+
+    return _setup_datasets(*(("AmazonReviewFull",) + args), **kwargs)
+
+
+def generate_imdb_data(key, extracted_files):
+    data_set = []
+    for fname in extracted_files:
+        if 'urls' in fname:
+            continue
+        elif key in fname and ('pos' in fname or 'neg' in fname):
+            with io.open(fname, encoding="utf8") as f:
+                label = 1 if 'pos' in fname else 0
+                data_set.append((label, f.read()))
+    return data_set
+
+
+def IMDB(root='.data'):
```

Comment:
This is the only function that accepts an explicit "root" argument.

In general, let's think about how this could be something more generic than a path. In the future users might want to pass other [file like objects](https://docs.python.org/3/glossary.html#term-file-object) as a source for dataset construction.